libfive/stdlib/stdlib.cpp

```diff
@@ -4,12 +4,16 @@
  *  DO NOT EDIT BY HAND!
  *  This file is automatically generated from libfive/stdlib/libfive_stdlib.h
  *
- *  It was last generated on 2022-02-09 19:39:14 by user bramp
+ *  It was last generated on 2022-02-10 18:56:25 by user bramp
  */
 #include "libfive_stdlib.h"
 #include "stdlib_impl.hpp"
 
 using namespace libfive;
+
```

Comment:
This shouldn't be necessary, because this is a C++ file (rather than a header which could be included in either C or C++)

Reply:
ah yes, I can drop the #ifdef, but the extern "C" {} I think has to stay, as that's how it's defined in the header file

Reply:
Are you sure?  [This answer](https://stackoverflow.com/questions/2168241/is-it-required-to-add-extern-c-in-source-file-also) suggests it _only_ needs to be in the header, not in the source; [`libfive.cpp`](https://github.com/libfive/libfive/blob/master/libfive/src/libfive.cpp) also doesn't use it, but defines functions that are callable from the C API.

Reply:
Oh interesting, it does say

> A function can be declared without a linkage specification after an explicit linkage specification has been seen; the linkage explicitly specified in the earlier declaration is not affected by such a function declaration.

but the answerer goes on to say:

> I'm not convinced it's good practice though, since there's the potential for the linkage specifications to diverge by accident (if, for example, the header file containing the linkage specification isn't included in the implementing file). I think it's better to be explicit in the implementation file.

I would agree with that, since it just seems more explicit that the header matches the implementation... but I'm happy to drop it.